js/data/bucket/symbol_bucket.js

```diff
@@ -184,6 +192,12 @@ class SymbolBucket {
             // It's better to place labels on one long line than on many short segments.
             this.features = mergeLines(this.features);
         }
+
+        // set up this mapping to allow for constant-time lookups in place()
+        this.featuresByIndex = {};
```

Comment:
Rather than create a separate index, can we attach a reference to the feature properties to the `symbolInstance`?